SConstruct

```diff
@@ -2888,6 +2888,23 @@ BOOST_AUTO_TEST_CASE(f)
 		if register_runtime_test_link_targets:
 			self.check_boost_test(context)
 
+	# dylibbundler is used to embed libraries into macOS app bundles,
+	# however it's only available on macOS.  While required for macOS
+	# builds, builds should not fail on other operating systems if
+	# it's absent.
+	@_custom_test
+	def _check_dylibbundler(self,context):
+		context.Display('%s: checking whether dylibbundler is installed...' % self.msgprefix)
+		if sys.platform != 'darwin':
+			context.Result('n/a')
+			return True
+		dylibbundler_output = os.popen('dylibbundler -h').read()
+		if 'dylibbundler is a utility' not in dylibbundler_output:
+			context.Result('no')
+			raise SCons.Errors.StopError('dylibbundler is required for compilation on macOS\n')
```

Comment:
I don't think this is quite accurate.  As I read the code flow, dylibbundler is only required `if not self.user_settings.macos_add_frameworks`.  Users who leave frameworks enabled will not attempt to run `dylibbundler`.

Also, as a minor point, no trailing newline is needed for the string passed to `StopError`.

Reply:
That's a good point about the framework situation.  I'll fix that.